internal/service/source.go

```diff
@@ -125,6 +125,88 @@ func (s *SourceService) DeleteSource(ctx context.Context, id uuid.UUID) error {
 	return nil
 }
 
+func (s *SourceService) UpdateSource(ctx context.Context, id uuid.UUID, form mappers.SourceUpdateForm) (*model.Source, error) {
+	ctx, err := s.store.NewTransactionContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	source, err := s.store.Source().Get(ctx, id)
+	if err != nil {
+		_, _ = store.Rollback(ctx)
+		if errors.Is(err, store.ErrRecordNotFound) {
+			return nil, NewErrSourceNotFound(id)
+		}
+		return nil, err
+	}
+
+	// Update source basic fields
+	if form.Name != nil {
+		source.Name = *form.Name
+	}
+
+	// Update labels if provided
+	if form.Labels != nil {
+		// Convert API labels to model labels using the mapper
+		labels := mappers.LabelsFromApi(source.ID, form.Labels)
+
+		// Update labels using the new UpdateLabels function
+		if err := s.store.Label().UpdateLabels(ctx, source.ID, labels); err != nil {
```

Comment:
This is a bit overengineered. Gorm can update the labels by itself. You don't need to update it.
Please see https://github.com/kubev2v/migration-planner/blob/52711e7c6add6ce60a2bac7a093896efb48b987a/internal/store/source_test.go#L369